src/python/WMCore/WMRuntime/Scripts/SetupCMSSWPset.py

```diff
@@ -647,27 +650,33 @@ def handleEnforceGUIDInFileName(self):
         if self.crabPSet:
             return
 
-        # only check this if we are in the first step of a workflow as output files of chained processing
-        # steps will not have correct GUID information
-        if self.step.data._internal_name != "cmsRun1":
-            self.logger.info("Not evaluating enforceGUIDInFileName parameter for step %s",
+        # only check this if we are in the first step of a workflow or there is a secondary input source
+        # output files of chained processing steps will not have correct GUID information
+        if self.step.data._internal_name != "cmsRun1" and not secondaryInput:
```

Comment:
`or`?

Reply:
I thought about this one for awhile yesterday. I think it's `and`.

We need to bail out for cases where step is cmsRun2 or higher, but we shouldn't if we are evaluating a secondary input (where afaik the step shouldn't matter).

Reply:
with AND:
cmsRun1 and secondary:  False
cmsRun1 and non-secondary:  False
cmsRun2 and secondary:  False
cmsRun2 and non-secondary:  True

with OR:
cmsRun1 or secondary:  False
cmsRun1 or non-secondary:  True
cmsRun2 or secondary:  True
cmsRun2 or non-secondary:  True

Indeed it twisted my mind as well (and still does not deliver the exact outcome we want) :) How about evaluating only the cmsRun number, keeping the rest of the logic you have already put in place.

Reply:
Wait a sec, I'm going to switch from True/False to Don't Evaluate/Evaluate. Since True really means don't evaluate (return) in this case. Don't we want...
with AND:
cmsRun1 and secondary: Evaluate
cmsRun1 and non-secondary: Evaluate
cmsRun2 and secondary: Evaluate
cmsRun2 and non-secondary: Don't Evaluate

Anyways it might be too confusing to keep like it is anyways... :)